actionpack/lib/action_controller/metal/conditional_get.rb

```diff
@@ -77,18 +77,16 @@ def etag(&etagger)
     #
     #   before_action { fresh_when @article, template: 'widgets/show' }
     #
-    def fresh_when(record_or_options, additional_options = {})
-      if record_or_options.is_a? Hash
-        options = record_or_options
-        options.assert_valid_keys(:etag, :last_modified, :public, :template)
-      else
-        record  = record_or_options
-        options = { etag: record, last_modified: record.try(:updated_at) }.merge!(additional_options)
+    def fresh_when(record = nil, etag: record, last_modified: nil, public: false, template: nil)
+      last_modified ||= record.updated_at if record
```

Comment:
@kaspth :tada: I was going to write this PR myself but you did almost everything! :+1: 

Regarding the `updated_at`, you cannot safely remove the `try` if you want this change to be backward-compatible